frontend/src/components/Buttons/SortButton.vue

```diff
@@ -1,63 +1,56 @@
 <template>
   <v-menu offset-y>
-    <template #activator="{ on, attrs }">
+    <template #activator="{ props }">
       <v-btn
         v-if="!$vuetify.display.smAndDown"
         class="my-2"
         variant="text"
         rounded
         :disabled="disabled"
-        v-bind="attrs"
-        v-on="on">
-        {{ $t('sortByType', { type: items[model].name }) }}
+        v-bind="props">
+        {{
+          $t('sortByType', {
+            type:
+              model == 0
+                ? items[0].title
+                : items.find((x) => x.value == model[0]).title
+          })
+        }}
         <v-icon end>
           <i-mdi-menu-down />
         </v-icon>
       </v-btn>
-      <v-btn
-        v-else
-        :disabled="disabled"
-        class="my-2"
-        icon
-        v-bind="attrs"
-        v-on="on">
+      <v-btn v-else :disabled="disabled" class="my-2" icon v-bind="props">
         <v-icon>
           <i-mdi-sort-alphabetical-ascending />
         </v-icon>
       </v-btn>
     </template>
-    <v-list dense>
-      <v-list-group
-        v-model="model"
-        @change="$emit('change', items[model].value)">
-        <v-list-item v-for="item in items" :key="item.value">
-          <v-list-item-title>{{ item.name }}</v-list-item-title>
-        </v-list-item>
-      </v-list-group>
+    <v-list
+      :items="items"
+      class="filter-content"
+      v-model:selected="model"
+      @update:selected="$emit('change', model.value)">
     </v-list>
   </v-menu>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { ref, defineProps } from 'vue';
+import { useI18n } from 'vue-i18n';
 
-export default defineComponent({
-  props: {
-    disabled: {
-      type: Boolean,
-      required: false,
-      default: false
-    }
-  },
-  data() {
-    return {
-      items: [
-        { name: this.$t('name'), value: 'SortName', order: false },
-        { name: this.$t('rating'), value: 'CommunityRating', order: false },
-        { name: this.$t('releaseDate'), value: 'PremiereDate', order: false }
-      ],
-      model: 0
-    };
+const { t } = useI18n();
+const props = defineProps({
+  disabled: {
+    type: Boolean,
+    required: false,
+    default: false
   }
 });
+const items = [
+  { title: t('name'), value: 'SortName', order: false },
+  { title: t('rating'), value: 'CommunityRating', order: false },
+  { title: t('releaseDate'), value: 'PremiereDate', order: false }
+];
+var model = ref(0);
```

Comment:
```suggestion
const model = ref(0);
```

This is not strictly enforced by the linter, but please let's try to stick to define all the refs and variables at the top of the script setup section